packages/ra-tree-ui-materialui/src/NodeView.js

```diff
@@ -36,53 +36,66 @@ export class NodeView extends Component {
         basePath: PropTypes.string.isRequired,
         children: PropTypes.node,
         classes: PropTypes.object,
-        node: PropTypes.object.isRequired,
+        item: PropTypes.object.isRequired,
         resource: PropTypes.string.isRequired,
     };
 
-    handleClick = event => {
-        event.persist();
-        // This ensure clicking on a button does not collapse/expand a node
-        // When clicking on the form (empty spaces around buttons) however, it should
-        // propagate to the parent
-        if (!event.target.matches(`.${CONTAINER_CLASS}`)) {
-            event.stopPropagation();
-        }
+    handleCollapse = () => {
+        const { onCollapse, item } = this.props;
+        onCollapse(item.id);
+    };
+
+    handleExpand = () => {
+        const { onExpand, item } = this.props;
+        onExpand(item.id);
     };
 
     render() {
         const {
             actions,
-            basePath,
             children,
             classes,
-            node,
-            resource,
+            item,
+            provided,
+            onCollapse,
+            onExpand,
             ...props
         } = this.props;
 
         return (
-            <div
-                className={classNames(CONTAINER_CLASS, classes.root)}
-                onClick={this.handleClick}
-                {...sanitizeRestProps(props)}
-            >
-                {Children.map(children, field =>
-                    field
-                        ? cloneElement(field, {
-                              basePath: field.props.basePath || basePath,
-                              record: node.record,
-                              resource,
-                          })
-                        : null
-                )}
-                {actions &&
-                    cloneElement(actions, {
-                        basePath,
-                        record: node.record,
-                        resource,
-                    })}
-            </div>
+            <Card className={classes.root}>
```

Comment:
not a big fan of the extra shadow caused by a Card. Is it really necessary?

Reply:
It allows to see the different items more clearly imo. What do you suggest ?